pageserver/src/keyspace.rs

```diff
@@ -10,6 +10,7 @@ pub struct KeySpace {
     /// Contiguous ranges of keys that belong to the key space. In key order,
     /// and with no overlap.
     pub ranges: Vec<Range<Key>>,
+    pub start: Key,
```

Comment:
Is this different from `ranges[0].start`? Could a field comment be added?

Reply:
If it's not, I don't understand how the changes in commit achieve what they say they would do.

Reply:
It is actually related to different problem.
Which is deeply related and leads to the same problem: old layers are not deleted by GC.
It is explained here:
https://neondb.slack.com/archives/C04DGM6SMTM/p1678121867070479
I will extract it in separate PR.